client/config/dfget.go

```diff
@@ -234,16 +234,9 @@ func (cfg *ClientOption) checkOutput() error {
 	}
 
 	f, err := os.Stat(cfg.Output)
-	// when recursive download, need a directory
-	if cfg.Recursive {
-		if err == nil && !f.IsDir() {
-			return fmt.Errorf("path[%s] is file but requires directory path", cfg.Output)
-		}
-	} else {
-		// when not recursive download, need a file
-		if err == nil && f.IsDir() {
-			return fmt.Errorf("path[%s] is directory but requires file path", cfg.Output)
-		}
+	// when not recursive download, need a file
+	if !cfg.Recursive && err == nil && f.IsDir() {
```

Comment:
In my opnion, here we shouldn't return an error, user's will feel frustated
Just use the orignal file name.

Reply:
User's dfget download request will be handled by following methed first.  
https://github.com/dragonflyoss/Dragonfly2/blob/7439b0e6c57d810b51ad0ba575b217ea88c78402/client/config/dfget.go#L171
If user not specify output with '-O' flag, then the output will be added according to the url